config.json

```diff
@@ -342,6 +354,24 @@
       "values": [],
       "defaultValue": ""
     },
+    "OptionalToolSource": {
```

Comment:
These are the same as the `EnableProfileGuidedOptimization` and `OptimizationDataDir` properties; the only difference is they are secrets, so they will need to be passed in from CI/VSTS or from users manually. VSTS support requires changes to not-checked-in pipeline definitions. CI support is coming in later PR(s). I want to go ahead with this PR before the CI changes are ready so that I can unblock manual IBC builds, which other work is dependent upon.

They *can* be passed directly to msbuild after `--`, but as I also noted above, that is much more fragile. No other parameters currently used in CI rely on `--`, which I think is a good thing. I'd really much rather add these settings to config.json for now, and remove them together with the non-secret ones once they're available in a shared config.